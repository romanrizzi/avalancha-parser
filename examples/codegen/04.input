
fun and_
  True  , True  -> True
  True  , False -> False
  False , True  -> False
  False , False -> False

fun or_
  True  , True  -> True
  True  , False -> True
  False , True  -> True
  False , False -> False

fun siguiente
  Norte -> Este
  Este  -> Sur
  Sur   -> Oeste
  Oeste -> Norte

print and_(True, True)
print and_(True, False)
print and_(False, True)
print and_(False, False)

print or_(True, True)
print or_(True, False)
print or_(False, True)
print or_(False, False)

print Q(
        siguiente(Norte),
        siguiente(siguiente(Norte)),
        siguiente(siguiente(siguiente(Norte))),
        siguiente(siguiente(siguiente(siguiente(Norte))))
      )

