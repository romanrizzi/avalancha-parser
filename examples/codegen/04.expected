True
False
False
False
True
True
True
False
Q(Este, Sur, Oeste, Norte)
